eclipse/src/saros/ui/widgets/viewer/project/BaseResourceSelectionComposite.java

```diff
@@ -386,223 +598,200 @@ protected void redoSelection() {
-          }
+    List<IContainer> sanitizedBaseResources =
+        determineBaseResources(new ArrayList<>(sanitizedResources));
 
-          return true;
-        }
-      };
+    setSelectedResourcesInternal(sanitizedBaseResources);
+  }
 
   /**
-   * Filter out any non-project resources.
+   * Returns the base resources contained in the given list of resources.
    *
-   * <p>Dirty workaround to restrict the menu to project elements for the migration period where
-   * partial sharing was removed but the reference point logic (and the subsequent rework of the
-   * sharing UI) has not been introduced yet.
+   * <p>In the returned list, it is guaranteed that none of the resources contained in the set is a
+   * child resource of another resource in the set.
+   *
+   * @param resources the list of resources
+   * @return the base resources contained in the given list of resources
    */
-  // TODO remove one reference point sharing has been introduced
-  protected ViewerFilter nonProjectFilter =
-      new ViewerFilter() {
-        @Override
-        public boolean select(Viewer viewer, Object parentElement, Object element) {
-          return element instanceof IProject;
+  private List<IContainer> determineBaseResources(List<IContainer> resources) {
+    List<IContainer> baseResources = new ArrayList<>();
+
+    resources.sort(
+        (r1, r2) -> {
+          IPath path1 = r1.getFullPath();
+          IPath path2 = r2.getFullPath();
+
+          // sort in ascending order
+          return Integer.compare(path1.segmentCount(), path2.segmentCount());
+        });
+
+    for (IContainer resource : resources) {
+      boolean isChild = false;
+
+      for (IContainer baseResource : baseResources) {
+        isChild = isChildResource(baseResource, resource);
+
+        if (isChild) {
+          break;
         }
-      };
+      }
 
-  @Override
-  protected CheckboxTreeViewer createViewer(int style) {
-    return new CheckboxTreeViewer(new Tree(this, style));
-  }
+      if (!isChild) {
+        baseResources.add(resource);
+      }
+    }
 
-  @Override
-  protected void configureViewer(CheckboxTreeViewer viewer) {
-    viewer.setContentProvider(new WorkbenchContentProvider());
-    viewer.setLabelProvider(new WorkbenchLabelProvider());
-    viewer.setUseHashlookup(true);
-    viewer.setSorter(new WorkbenchItemsSorter());
+    return baseResources;
   }
 
   /**
-   * Sets the currently selected {@link IResource}s.
+   * Sets selects the given resources. This selection is handled as though it were made by the user,
+   * meaning all necessary listeners and internal handlers are called.
    *
-   * @param resources
+   * <p><b>NOTE:</b> This method expects that the list of given resources only contains base
+   * resources, i.e. that none of the contained resources is a child resource of another contained
+   * resource. Furthermore, it expects that all given resources are containers.
```

Comment:
It takes a list of IContainers. Any chance that a resource is not a container ?

Reply:
Done.